jar-infer/jar-infer-lib/src/test/java/com/uber/nullaway/jarinfer/EntriesComparator.java

```diff
@@ -104,4 +106,40 @@ public static boolean compareEntriesInAars(String aarFile1, String aarFile2) thr
     }
     return jar1Entries.equals(jar2Entries);
   }
+
+  private static String readManifestFromJar(String jarfile) throws IOException {
+    JarFile jar = new JarFile(jarfile);
+    ZipEntry manifestEntry = jar.getEntry("META-INF/MANIFEST.SF");
+    if (manifestEntry == null) {
+      return "";
+    }
+    StringBuilder stringBuilder = new StringBuilder();
+    BufferedReader bufferedReader =
+        new BufferedReader(new InputStreamReader(jar.getInputStream(manifestEntry), "UTF-8"));
+    String currentLine;
+    while ((currentLine = bufferedReader.readLine()) != null) {
+      // Ignore empty new lines
+      if (currentLine.trim().length() > 0) {
+        stringBuilder.append(currentLine);
+      }
+    }
+    return stringBuilder.toString();
+  }
+
+  /**
+   * Compares the META-INF/MANIFEST.SF file in the given 2 jar files. We ignore empty newlines.
+   *
+   * @param jarFile1 Path to the first jar file.
+   * @param jarFile2 Path to the second jar file.
+   * @return True iff the MANIFEST.SF files in the two jar files exist and are the same.
```

Comment:
This is going to return true if both jar files do not have MANIFEST.SF, because you are returning an empty string in that case.

Reply:
Right! Do you think this is an error in the documentation? e.g. comparing the manifest files of two jars without manifest files should be `true` by vacuity. Or should we just throw `IllegalArgumentException` or something like that instead? The first is probably more elegant, but the second seems "safer", so I am leaning that way for test code...

Reply:
Do you think there could be a problem if both Jars do not have MANIFEST.SF? I can't think of any problems with that.

I am okay with keeping the code as is now, and updating the documentation to reflect this.

Reply:
Actually, I did the `IllegalArgumentException` check and found an error on my test because of it. The name of the file is `MANIFEST.`**MF** ... whoops!

The fixed test actually fails because the pre-JarSigner pre-JarInfer jar has no `MANIFEST.MF`, but the one at the end has one with jus the line: "Manifest-Version: 1.0" ...

Reply:
Oh, okay then. Lets keep it that way.

I actually noticed the `.SF` in the file name in your code, but I assumed that it is for the signed version. I should've checked and confirmed that. Sorry.

Reply:
Yeah, the annoying part is there is an `.SF` file, but that's separate from `MANIFEST.MF`, which is why I got confused. Test and code seem sane now. Sorry for the force push, had to do it to rebase :)